### PR TITLE
Replace $ symbol with € symbol on some layouts

### PR DESCRIPTION
**Pull request by trev0r-STA**
Modified layout:
-Dutch (dutch and belgium dutch)
-French (France, belgium)
-Finnish
-Italian (Italia only)
-Portugese (Portugal only)
-Spanish (basque, catalan, spanish)

I've modified method.xml and the differents xml disposition. Not sure if I've modified all the files. Please feel free to help

Have a nice day

### CODE REVIEW

File: app/src/main/res/xml/method.xml [hunk 1]
```diff
@@ -232,7 +232,7 @@
             android:subtypeId="0xd2e520d5"
             android:imeSubtypeLocale="ca"
             android:imeSubtypeMode="keyboard"
-            android:imeSubtypeExtraValue="KeyboardLayoutSet=spanish,AsciiCapable,EmojiCapable"
+            android:imeSubtypeExtraValue="KeyboardLayoutSet=spanish_euro,AsciiCapable,EmojiCapable"
```
Comment (MajeurAndroid) [inline app/src/main/res/xml/method.xml:235]: I'm not sure you have to change anything in a such high level.

**Comment by MajeurAndroid** [comment]
Thanks for you contribution. I'm not sure about the scope you modified tough. Adding an euro symbol should be done in keyboard keys definitions as you might have seen, I don't really get why you did duplicate all these keyboards. Where are you trying to add the euro symbol ? It should already be there for European countries you adressed.

**Comment by trev0r-STA** [comment]
Well
For what I understand. Layouts ar ebuild like this:
method.xml  --> define keyboard layout per language (XXXX)
-->keyboard_layout_set_XXXX.xml --> The differents types of keyboards for this language (phone/symbol/numbers/alphabet)
---->kbd_XXXX.xml --> Which rows definition to load for the keyboard
------>rows_XXXX.xml --> The differents rows of the keyboard (symbols, first letters row, second, etc)
--------> rowkeys_XXXX.xml --> The definition of each key for this row. And this is where I need to replace $ with €
As definitions are used in multiple layouts, I need to rename all the files described above, for changing only for one language.
For example:  spanish keyboard is derivate from the azerty keyboard plus the ñ touch. But it's used for spanish, catalan and also spanish for United States and Latin America, countries that  don't use euro. Same thing for finnish which is the same keyboard as danish, norwegian, swedish... And they don't use €. 
Last but not least: if you look for italian, they used the qwerty keyboard. I can't change the definition of the key, otherwise all the qwerty keyboards will have the € symbol
Am I clear ? 😉

**Comment by MajeurAndroid** [comment]
Ok i hunderstand what you wanted to do. I think a switch case on Locale can be done in symbols keys definition, which will have a smaller footprint.
I think I saw a commit you can get inspiration from. Here it is : https://github.com/LineageOS/android_packages_inputmethods_LatinIME/commit/bd6fcb304dfb61146d7ea7d3e10c4a057b157c8e

**Comment by trev0r-STA** [comment]
Well. I read your link. The same file exists in the project, with the correct values... But it doesn't seem to be used or there is a bug somewhere, and I don't have the capacity to find where the bug is.
What should I do ? Remove my commit and delete the PR (if I can) ?

**Comment by Garettman** [comment]
> After dive a little bit more into the code, the file key_styles_currency.xml is never use. All the european countries which have the € symbol on there layouts are modified layouts as I've done

Yes, that's what i thought too after some investigation. I may have found a useful file at `app/src/main/java/org/dslul/openboard/inputmethod/keyboard/internal/KeyboardTextsTable.java /` after a little inspection of `app/src/main/res/xml/method.xml` but I tried modifying it by adding ` * morekeys_d */ "\u20AC",` to add Euro symbol while pressing d and it doesn't work, don't know why...

**Comment by MajeurAndroid** [comment]
Have a look at the README.md, I have added instructions to edit this particular file.
I'll investigate your PR over the weekend.

**Comment by MajeurAndroid** [comment]
@trev0r-STA after some investigations, I did not understand what you're trying to achieve. From what I see, each locale you mentionned already has the € symbol. It is done through the key_styles_currency.xml config file to avoid defining an entire keyboard each time. It is used for every keyboard through the rows_symbols.xml.

By the way, I updated the list of countries that should have the euro symbol as of September 2019 from the LineageOS commit I mentionned  before. Changes are here b868e15.

**Comment by trev0r-STA** [comment]
@MajeurAndroid Anyway, I don't understand how the code works, for me java is like chinese, so... However, the € symbol did not display on the french layout and android 7.1.2...

**Pull request closed without merging**